src/PostgREST/Response/OpenAPI.hs

```diff
@@ -307,20 +308,23 @@ makePathItem t = ("/" ++ T.unpack tn, p $ tableInsertable t || tableUpdatable t
     ref = Ref . Reference
 
 makeProcPathItem :: ProcDescription -> (FilePath, PathItem)
-makeProcPathItem pd = ("/rpc/" ++ toS (pdName pd), pe)
+makeProcPathItem pd = ("/rpc/" ++ toS (pdName pd), pe $ pdVolatility pd == Volatile)
```

Comment:
I have a doubt about this. The only way to check the volatility of a function is through this parameter, but PostgreSQL makes all the functions `VOLATILE by default so the `get` param won't show for those functions even if the request using `GET` is allowed. 

If we go with this route the user will be obligated to specify the volatility of a function in order for the OpenAPI output to be accurate (needs to be mentioned in the docs). Or we could add the `get` parameter disregarding the volatility and risk getting `405 Method Not Allowed` even when `get` is present in the OpenAPI output.

Reply:
> so the get param won't show for those functions even if the request using GET is allowed.

You mean the function will still work through GET because it will be compatible with a read-only transaction right?

> Or we could add the get parameter disregarding the volatility and risk getting 405 Method Not Allowed even when get is present in the OpenAPI output.

Yeah, I guess this would be most correct because it's what we actually support.

---

If we did https://github.com/PostgREST/postgrest/issues/1826 and indeed disable GET/HEAD for volatile functions, then conditionally enabling GET would be accurate. But this can be done later and it'd be a breaking change.